test/string.js

```diff
@@ -2580,7 +2580,7 @@ describe('string', function () {
 
         it('validates combination of isoDate, min, max, regex and required', function (done) {
 
-            var rule = Joi.string().isoDate().min(17).max(23).regex(/Z$/).required();
+            var rule = Joi.string().isoDate().min(10).max(23).regex(/[^T]$|Z$|[+-][^a-zA-Z].*$/).required();
```

Comment:
It looks like you've fundamentally changed some of the old tests why not just create new ones? Since this library is used across many projects we should make sure it functions like it used to as well as for the new functionality.